scripts/src/java/org/oppia/android/scripts/RegexPatternValidationCheck.kt

```diff
@@ -0,0 +1,232 @@
+package org.oppia.android.scripts
+
+import java.io.File
+import java.io.FileInputStream
+import org.oppia.android.app.model.FilenameChecks
+import org.oppia.android.app.model.FilenameCheck
+import org.oppia.android.app.model.FileContentChecks
+import org.oppia.android.app.model.FileContentCheck
+import org.oppia.android.scripts.ScriptResultConstants
+
+class RegexPatternValidationCheck {
+  companion object {
+    @JvmStatic
+    fun main(vararg args: String) {
+
+      val repoPath = args[0] + "/"
+
+      val allowedDirectories: MutableList<String> = ArrayList()
```

Comment:
More Kotlin idiomatic: ``val allowedDirectories = mutableListOf<String>()``

However, given how this is being initialized could you maybe instead do:

```kotlin
val allowedDirectories = args.drop(1)
```

Now it's immutable, and avoids needing to do the extra for-loop.

Reply:
This seems neat. Updated.